### PR TITLE
RDHPCS Hera Rocky-8 support

### LINKED ISSUE #913: Update RDHPCS Hera to Rocky-8 compliance
**Description**

RDHPCS Hera is being upgraded to Linux [Rocky version 8](https://rockylinux.org/). As a result all build applications, namely module files must be updated accordingly.

**Required Functionality**

A UFS based applications used by the UFS global-workflow must be updated to Rocky-8.

**Proposed Solution**

Update all UFS global-workflow applications to Rocky-8 compatability.

**Definition of Completion**

The global-workflow and the associated UFS applications reproduce RHEL 8 instances. Further, all unit-tests pass with success.

**Related Issues**

Global workflow issue [#2329](https://github.com/NOAA-EMC/global-workflow/issues/2329)

### PR DESCRIPTION
**Pull request by HenryRWinterbottom**
## DESCRIPTION OF CHANGES: 

Module files and regression tests updated for the Rocky-8 stack upgrade.

## TESTS CONDUCTED: 

- [x] Compile branch using Intel on Hera. Done using 3a5fb60.
- [x] Compile branch on Hera using GNU. Done using 3a5fb60.
- [x] Run unit tests Hera. Done using 3a5fb60. All tests passed.
- [x] Run relevant consistency tests on Hera. Done using 3a5fb60. All tests passed except one `chgres_cube` test. But this is not a problem. See below for details.

Tests were performed only for RDHPCS Hera as it is the only machine impacted.

## ISSUE: 

Resolves #913 
References [#2329](https://github.com/NOAA-EMC/global-workflow/issues/2329)

### CODE REVIEW

File: modulefiles/build.hera.intel.lua [hunk 1]
```diff
@@ -49,7 +49,7 @@ load(pathJoin("libpng", png_ver))
 netcdf_c_ver=os.getenv("netcdf_c_ver") or "4.9.2"
 load(pathJoin("netcdf-c", netcdf_c_ver))
 
-netcdf_fortran_ver=os.getenv("netcdf_fortran_ver") or "4.6.1"
+netcdf_fortran_ver=os.getenv("netcdf_fortran_ver") or "4.6.0"
```
Comment (DavidHuber-NOAA) [inline modulefiles/build.hera.intel.lua]: NetCDF version should be 4.6.1.

```suggestion
netcdf_fortran_ver=os.getenv("netcdf_fortran_ver") or "4.6.1"
```

File: modulefiles/build.hera.gnu.lua [hunk 2]
```diff
@@ -5,7 +5,7 @@ Load environment to compile UFS_UTILS on Hera using Gnu
 hpss_ver=os.getenv("hpss_ver") or ""
 load(pathJoin("hpss", hpss_ver))
 
-prepend_path("MODULEPATH", "/scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.6.0/envs/unified-env/install/modulefiles/Core")
+prepend_path("MODULEPATH", "/scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/modulefiles/Core")
 -- For openmpi:
```
Comment (GeorgeGayno-NOAA) [inline modulefiles/build.hera.gnu.lua]: @DavidHuber-NOAA - you added this MODULEPATH for openmpi at 892b693. If I comment it out, the branch still compiles. Is this line still needed? If so, does it work for Rocky8?

Comment (DavidHuber-NOAA) [reply modulefiles/build.hera.gnu.lua]: @GeorgeGayno-NOAA Good catch.  It seems the SAs have installed a version of `openmpi` in the standard stack, so this line should be removed.

@HenryWinterbottom-NOAA Can you please delete line 10:
```
prepend_path("MODULEPATH", "/scratch1/NCEPDEV/jcsda/jedipara/spack-stack/modulefiles")
```

**Comment by GeorgeGayno-NOAA** [comment]
The `c96.fv3.netcdf2wam` component of the `chgres_cube` regression test suite failed using 3a5fb60. Not sure I can explain why. The differences from the baseline were very small, however. So, this does not indicate a problem.  Here are the tile 1 differences:

```
+ nccmp -dmfqS out.atm.tile1.nc /scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/chgres_cube/baseline_data/c96_fv3_netcdf2wam/out.atm.tile1.nc
Variable Group Count         Sum      AbsSum          Min         Max       Range        Mean      StdDev
zh       /       112    0.257812     2.19531     -0.03125     0.03125      0.0625   0.0023019   0.0211469
t        /       117  0.00175476   0.0108795 -0.000213623 0.000305176 0.000518799  1.4998e-05 0.000107299
spo      /       729 6.04081e-06 4.45471e-05 -9.23872e-07 9.23872e-07 1.84774e-06 8.28643e-09 1.77342e-07
spo2     /      1859 4.17668e-06 3.18948e-05 -5.51343e-07 6.70552e-07  1.2219e-06 2.24673e-09 3.95506e-08
```

Will recommend merging.

**Review by GeorgeGayno-NOAA — APPROVED** [review]

**Pull request merged**